routing/router.go

```diff
@@ -1834,6 +1842,21 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 					)
 				}
 
+				// Update routing hints.
+				uBaseFee := lnwire.MilliSatoshi(update.BaseFee)
+				uFeeRate := lnwire.MilliSatoshi(update.FeeRate)
+				uMinHTLC := lnwire.MilliSatoshi(update.HtlcMinimumMsat)
+				uTimeLockDelta := update.TimeLockDelta
+				failedChanID := failedEdge.channelID
+				paySession.UpdateEdgePolicy(
+					&channeldb.ChannelEdgePolicy{
+						ChannelID:                 failedChanID,
+						MinHTLC:                   uMinHTLC,
+						TimeLockDelta:             uTimeLockDelta,
+						FeeBaseMSat:               uBaseFee,
+						FeeProportionalMillionths: uFeeRate,
+					})
+
```

Comment:
You've added it in here, but for other cases where `r.applyChannelUpdate` is called directly, the payment session isn't updated.

Reply:
My idea is that the hints are updated only when a retry is aimed at. It now handles all those cases (not only insufficient fees). Otherwise the vertex or channel is removed completely, and there is no need for the update. I thought that is the whole purpose of defining processChannelUpdateAndRetry (which includes the paySession in its closure). 
Said in another way, it doesn't seem right to me to plug it into applyChannelUpdate, exactly because this function is called for changing database edges/vertexes (and don't have paySession in closure).

Reply:
Yes, I think you're right. All other calls to `applyChannelUpdate` prune immediately, so those edges will never show up in subsequent payment attempts anymore.